CleverPush/Source/CPAppBannerModuleInstance.m

```diff
@@ -279,38 +279,39 @@ - (BOOL)bannerTargetingAllowed:(CPAppBanner*)banner {
     }
 
     NSString* appVersion = [[NSBundle mainBundle].infoDictionary valueForKey:@"CFBundleShortVersionString"];
-    allowed = [self checkRelationFilter:appVersion compareWith:banner.appVersionFilterValue relation:banner.appVersionFilterRelation isAllowed:allowed];
+    allowed = [self checkRelationFilter:appVersion compareWith:banner.appVersionFilterValue relation:banner.appVersionFilterRelation isAllowed:allowed compareWithFrom:banner.fromVersion compareWithTo:banner.toVersion];
     return allowed;
 }
 
 #pragma mark - check the banner triggering allowed as per selected version match with app version or not.
-- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue relation:(NSString*)relation isAllowed:(BOOL)allowed {
+- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue relation:(NSString*)relation isAllowed:(BOOL)allowed compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo {
     return [self checkRelationFilter:value compareWith:compareValue compareWithFrom:compareValue compareWithTo:compareValue relation:relation isAllowed:allowed];
 }
 
 #pragma mark - check the banner triggering allowed as per selected version match with app version or not.
 - (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo relation:(NSString*)relation isAllowed:(BOOL)allowed {
+    
     if (relation == nil || compareValue == nil) {
         return allowed;
     }
     if (allowed && [relation isEqualToString:filterRelationType(CPFilterRelationTypeEquals)]) {
-        if (allowed && !CHECK_FILTER_EQUAL_TO(value, compareValue)) {
+        if (allowed && [value isEqualToVersion:compareValue] == NO) {
             allowed = NO;
         }
     } else if (allowed && [relation isEqualToString:filterRelationType(CPFilterRelationTypeGreaterThan)]) {
-        if (!CHECK_FILTER_GREATER_THAN(value, compareValue)) {
+        if ([value isEqualOrOlderThanVersion:compareValue] == YES) {
             allowed = NO;
         }
     } else if (allowed && [relation isEqualToString:filterRelationType(CPFilterRelationTypeLessThan)]) {
-        if (!CHECK_FILTER_LESS_THAN(value, compareValue)) {
+        if ([value isEqualOrNewerThanVersion:compareValue] == YES) {
```

Comment:
Actually we don't need `== YES` We can just do
`if ([value isEqualOrNewerThanVersion:compareValue]) {`

Also apply to all other conditions please.

Same with NO, just check if `!condition`